### PR TITLE
Add support for VueJs Framework (SFC)

### PR DESCRIPTION
**Pull request by LissetteIbnz**
This PR adds support for single-file [Vue.js](https://vuejs.org/) components.
The files with `.vue` extension allow to write html, javascript/typescript and styles in the same file.

### CODE REVIEW

**Comment by OriginalEXE** [comment]
Any chance someone could review this?

Thanks @LissetteIbnz for the PR!

**Comment by marcoscaceres** [comment]
Sorry to sit on this. We are trying to move new languages to their own repositories to avoid exactly these kinds of delays.

**Comment by jankal** [comment]
@marcoscaceres the heck, can we just get this merged until "the new way" is stable to toll into prod? It's quite annoying when nothing moves forward.

**Comment by sustained** [comment]
Looking forward to this being added. Any updates?

**Comment by ChangJoo-Park** [comment]
@LissetteIbnz pug is not working. It may needs support for pug.
Can you check?

File: test/detect/vue/default.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      :id="foo + 'baz'"
+      @click="onClick('hello')"
+      @click.prevent="onClick('hello')">
+      Hello
+    </span>
+  </div>
+</template>
+
+<template lang="pug">
+  <div>
```
Comment (ChangJoo-Park) [inline test/detect/vue/default.txt]: It need to be changed `div`. not `<div>`, also any other tags inside template using "pug"

Comment (LissetteIbnz) [reply test/detect/vue/default.txt]: The sample has been updated but does not implementat a specific highlight for pug 🤷‍♀️

Comment (ChangJoo-Park) [reply test/detect/vue/default.txt]: thanks :)

**Comment by marcoscaceres** [comment]
Ok, moved everything to the new repo and made @LissetteIbnz admin: 
https://github.com/highlightjs/highlightjs-vue/

@LissetteIbnz please feel free to publish on NPM and add any contributors. Also, if anyone here would like to volunteer, please let us know.

**Pull request closed without merging**